src/main/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImpl.java

```diff
@@ -65,14 +65,13 @@ void anonymiseCourtCaseEntity(UserAccountEntity userAccount, CourtCaseEntity cou
         courtCase.getDefenceList().forEach(defenceEntity -> anonymiseDefenceEntity(userAccount, defenceEntity));
         courtCase.getProsecutorList().forEach(prosecutorEntity -> anonymiseProsecutorEntity(userAccount, prosecutorEntity));
         courtCase.getHearings().forEach(hearingEntity -> anonymiseHearingEntity(userAccount, hearingEntity));
-        anonymiseAllEventsFromCase(userAccount, courtCase);
-
         courtCase.markAsExpired(userAccount);
         //This also saves defendant, defence and prosecutor entities
         tidyUpTransformedMediaEntities(userAccount, courtCase);
         auditApi.record(AuditActivity.CASE_EXPIRED, userAccount, courtCase);
         anonymiseCreatedModifiedBaseEntity(userAccount, courtCase);
         caseService.saveCase(courtCase);
+        anonymiseAllEventsFromCase(userAccount, courtCase);
```

Comment:
When does it actually save the anonymised event back to the database?

Reply:
Spring should auto save any modifications when we exit the transaction.
But for the avoidance of any doubt I have added a save call into the anonymise method